packages/shared/src/variants/drift.ts

```diff
@@ -11,6 +18,9 @@ export class DriftGo extends Baduk {
   private typedConfig: DriftGoConfig;
 
   constructor(config?: DriftGoConfig) {
+    if (config && !isGridBadukConfig(config)) {
+      throw Error("config for drift must be for board type grid");
+    }
     super(config);
     this.typedConfig = config ?? this.defaultConfig();
   }
```

Comment:
I think we can change the type of the board to a Grid here.

See
```
class Board {
  foo() {
    return "graph";
  }
}

class SpecificBoard extends Board {
  foo() { return "grid"; }
  bar() { return "asdf"; }
}

class GameParent {
  constructor(protected board: Board) {

  }

  public foo() {
    return this.board.foo();
  }
}

class GameChild extends GameParent {
  constructor(protected board: SpecificBoard) {
    super(board);
  }

  public bar() {
    return this.board.bar();
  }
}

const game = new GameChild(new SpecificBoard());
console.log(game.foo()); // prints "grid"
console.log(game.bar()); // prints "asdf"
```

Then we would need the `as Grid` type assertions.